pkg/tls/certRenewer.go

```diff
@@ -172,32 +197,59 @@ func (c *CertRenewer) WriteTLSPairToSecret(props CertificateProps, pemPair *PemP
 	logger := c.log.WithName("WriteTLSPair")
 
 	name := generateTLSPairSecretName(props)
+
+	depl, err := c.client.GetResource("", "Deployment", props.Namespace, config.KyvernoDeploymentName)
+
+	deplHash := ""
+	if err == nil {
+		deplHash = fmt.Sprintf("%v", depl.GetUID())
+	}
+
+	var deplHashSec string = "default"
+	var ok, managedByKyverno bool
+
 	secretUnstr, err := c.client.GetResource("", "Secret", props.Namespace, name)
-	if err != nil {
-		secret := &v1.Secret{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "Secret",
-				APIVersion: "v1",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      name,
-				Namespace: props.Namespace,
-				Labels: map[string]string{
-					ManagedByLabel: "kyverno",
-				},
+	if err == nil {
+		if label, ok := secretUnstr.GetLabels()[ManagedByLabel]; ok {
+			managedByKyverno = label == "kyverno"
+		}
+		deplHashSec, ok = secretUnstr.GetAnnotations()[MasterDeploymentUID]
+	}
+
+	secretPtr := &v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: props.Namespace,
+			Annotations: map[string]string{
+				MasterDeploymentUID: deplHash,
 			},
-			Data: map[string][]byte{
-				v1.TLSCertKey:       pemPair.Certificate,
-				v1.TLSPrivateKeyKey: pemPair.PrivateKey,
+			Labels: map[string]string{
+				ManagedByLabel: "kyverno",
 			},
-			Type: v1.SecretTypeTLS,
-		}
+		},
+		Data: map[string][]byte{
+			v1.TLSCertKey:       pemPair.Certificate,
+			v1.TLSPrivateKeyKey: pemPair.PrivateKey,
+		},
+		Type: v1.SecretTypeTLS,
+	}
 
-		_, err := c.client.CreateResource("", "Secret", props.Namespace, secret, false)
+	if err != nil {
```

Comment:
This is the error when getting the secret, you need to check the error type and process errors differently:
- if it's a NotFound error, create the secret
- if it's other types of errors, need to handle it differently